CodeEdit/Documents/WorkspaceCodeFileView.swift

```diff
@@ -30,26 +31,65 @@ struct WorkspaceCodeFileView: View {
                 }
                 return file.tabID == workspace.selectionState.selectedId
             }) {
-                if let codeFile = workspace.selectionState.openedCodeFiles[item] {
-                    CodeFileView(codeFile: codeFile)
-                        .safeAreaInset(edge: .top, spacing: 0) {
-                            VStack(spacing: 0) {
-                                BreadcrumbsView(file: item, tappedOpenFile: workspace.openTab(item:))
-                                Divider()
-                            }
-                        }
-                } else {
-                    Text("No Editor")
-                        .font(.system(size: 17))
-                        .foregroundColor(.secondary)
-                        .frame(minHeight: 0)
-                        .clipped()
+                if let fileItem = workspace.selectionState.openedCodeFiles[item] {
+                    switch fileItem.typeOfFile {
+                    case UTType.image:
+                        imageFileView(fileItem, for: item)
+                    case UTType.text:
+                        codeFileView(fileItem, for: item)
+                    default:
+                        otherFileView(fileItem, for: item)
+                    }
                 }
+            } else {
+                Text("No Editor")
+                    .font(.system(size: 17))
+                    .foregroundColor(.secondary)
+                    .frame(minHeight: 0)
+                    .clipped()
             }
         }
         .frame(maxWidth: .infinity, maxHeight: .infinity)
     }
 
+    @ViewBuilder
```

Comment:
I like how you split this code into methods! However, I think the `@ViewBuilder` wrapper before each method is unnecessary. If you always return only one view like `VStack` or `CodeFileView` - even with some modifiers - this wrapper is superfluous. If you are up to you can remove all of them from your methods. 
```suggestion
```